src/snowflake/snowpark/udaf.py

```diff
@@ -294,7 +294,7 @@ class UDAFRegistration:
         - :meth:`~snowflake.snowpark.DataFrame.join_table_function`
     """
 
-    def __init__(self, session: "snowflake.snowpark.session.Session") -> None:
+    def __init__(self, session: Optional["snowflake.snowpark.session.Session"]) -> None:
```

Comment:
Why is this Optional?

Reply:
All the decorators could either have the session = None or otherwise. Depending on that, 

```
    if session is None:
        udf_registration_method = UDFRegistration(session=session).register
    else:
        udf_registration_method = session.udf.register
```
When session is None, we cannot call session.udf.register, and hence need to use the UDFRegistration class directly. Initializing it requires a session, and since we are initializing it with a None session, the signature has been modified to accept the None session.